test/utils/IexecWrapper.ts

```diff
@@ -89,11 +89,21 @@ export class IexecWrapper {
             .then((tx) => tx.wait());
     }
 
+    async signAndSponsorMatchOrders(orders: IexecOrders) {
+        return this._signAndMatchOrders(orders, true);
+    }
+
+    async signAndMatchOrders(orders: IexecOrders) {
+        return this._signAndMatchOrders(orders, false);
+    }
+
     /**
      * @notice Before properly matching orders, this method takes care of
      * signing orders and depositing required stakes.
+     * A sponsor will be in charge of paying for the deal if `withSponsor` is enabled.
+     * Otherwise the requester will be in charge of paying for the deal.
      */
-    async signAndMatchOrders(orders: IexecOrders) {
+    private async _signAndMatchOrders(orders: IexecOrders, withSponsor: boolean) {
```

Comment:
We can use an optional argument `sponsor`
```suggestion
    private async _signAndMatchOrders(orders: IexecOrders, sponsor?: Signer) {
```

Reply:
sponsor is already available in class scope, hence it looks to me passing a signer does not make sense :thinking:

Reply:
I mean you can use optional arguments instead of true/false argument. They are used for this.
But no worries, never mind.

Reply:
Which optional arg do you suggest, `bool?`

Reply:
It's okey, let's keep it as it is.